hq/app/AppComponents.scala

```diff
@@ -52,14 +53,15 @@ class AppComponents(context: Context)
     }
   }
 
-  val cacheService = new CacheService(configuration, applicationLifecycle, environment)
+  private val cacheService = new CacheService(configuration, applicationLifecycle, environment)
+  private val googleAuthConfig = Config.googleSettings(httpConfiguration, configuration)
 
   override def router: Router = new Routes(
     httpErrorHandler,
-    new HQController(configuration, cacheService),
-    new SecurityGroupsController(configuration, cacheService),
-    new SnykController(configuration, configraun),
-    new AuthController(environment, configuration),
+    new HQController(configuration, cacheService, googleAuthConfig),
+    new SecurityGroupsController(configuration, cacheService, googleAuthConfig),
+    new SnykController(configuration, configraun, googleAuthConfig),
+    new AuthController(environment, configuration, googleAuthConfig),
```

Comment:
Minor note here but I'm not sure why it's become necessary to add `googleAuthConfig` to all the controllers? One note is that it might be better to pass a more tightly-focused object like a configured `AuthAction`, if that's what the `Controller` needs.

You could also go the full hog and create a `ControllerComponents` object if you want, like we do in Ophan: https://github.com/guardian/ophan/blame/fd7474fe9c632/dashboard/app/configuration/ApplicationComponents.scala#L71

Reply:
Beyond the scope of this change, but I will discuss with @adamnfish

Reply:
I agree that if all the required info is available in the config then passing the config would still be enough, but this does introduce a new piece of info that comes from the Play environment (the choice of secret and algorithm). I think doing this at the edge of the program and being explicit in the wiring makes sense. Certainly, creating `ControllerComponents` means you're still passing it into all the controllers and I'm not sure the indirection (and reduced specificity) that comes with using an intermediate data structure to hold the wiring gains much for us in this small program.